astropy/units/core.py

```diff
@@ -1480,7 +1480,7 @@ def __init__(self, st, doc=None, format=None, namespace=None):
         elif isinstance(st, tuple):
             if not len(st) == 2:
                 raise ValueError("st must be string, list or 2-tuple")
-            self._names = st[0] + st[1]
+            self._names = st[0] + [n for n in st[1] if n not in st[0]]
```

Comment:
I guess this works, but I am wondering if using `list(set(...))` would be simpler?